### ISSUE TITLE
Prompt model:-None Loaded error

### ISSUE THREAD

**Issue by Willxiam** [nathanlesage/local-chat#1]
When testing this out (thank you for making it available) on an older gaming pc
Windows 10 PC
16 gb ram
i5 processor
NVIDIA GeForce® GTX 1070 GAMING PCI Express video card 

**I receive "Prompt-None Loaded error" when I submit a query.**   

_Other Details_:
Release v0.4.0
manually download recommended models tinyllama-1.1b following links. Copied into model directory. 
Selected generic prompt template for model.
Downloading via Model manager did not seem to work. 

Have tried reloading model and the force reload tool in main menu window.

**Comment by nathanlesage** [comment]
So in 0.4.0 there was a bug where with new conversations you had to manually force reload the model once; in 0.5.0 this should happen automatically. So I'd recommend updating and then (just for good measure) force reloading the model (in the statusbar the reload button).

Does this solve the issue?

**Comment by Willxiam** [comment]
Updated to [Release v0.7.1](https://github.com/nathanlesage/local-chat/releases/tag/v0.7.1) and decided to again try and use the build in downloader,  I realized I was using the wrong download link.  See image below.   It continued not to work after updating to the latest version, force reloading model in both the manage model window and in the main window.   Along the bottom of application it says  "provider not initialized (unknown).  There is a button to force reload. WhenI click it it says "Could not reload model: None loaded"


Wrong  download link to copy. 
![image](https://github.com/nathanlesage/local-chat/assets/158068941/fb3b381e-b125-4b55-838e-b1cc49c47302)

**Comment by Willxiam** [comment]
Update: After rebooting it will now initialize the provider.  Showing " Model ready"  along the bottom.

**Comment by nathanlesage** [comment]
@Willxiam Exactly, I don't know why HuggingFace forces you to click on that download link to actually download the file, but that's the way it is …

Regarding stop generation button: While the model is generating, the "Force reload" button will turn into a "stop generating" button. If you click it, wait until the next token has been generated and the model will automatically stop generating.

**Comment by nathanlesage** [comment]
Mhmh, could be — I just today released a new version that also allows for CUDA support. I haven't yet enabled that flag, but the 0.8.0 should run exclusively on the CPU. This is not making most use of the system, BUT it should at least get you up and running. Still, the Node bindings are still in beta, so there will probably be more improvements over time.

**Comment by Willxiam** [comment]
thanks. I will give the latest release a go.  Cannot get to it until after work though.   After I got  localchat to recognize the model (not sure why it was not before) it would generate, but it would take quite some time.  If I remember correctly, about 15 to 30 minutes for a quick prompt.  And sometimes it would seem to get stuck in the attempt to generate.   I am assuming that running just on the CPU might be slowe3r unless there is some issue interfacing with my GPU> It has occurred to me that maybe my gpu has an updates set of drivers.  Or I could put a newer GPU in this system to get it working.

**Comment by Willxiam** [comment]
Ok  so I installed latest release and noticed that the reading of the model's meta data was much faster.  And it also seems like generation is faster and more stable.  
I also did some tests
7 minutes to generate answer to : What is the capital of France.?
Details:  generated  “the capital of france is Paris” at about the  450s.1s mark.
“the capital of france is Paris” 
Stopped generating at 534.9s 

12 minutes for What is 1+1?
generated 1+1= at about 330.5s 
got the correct answer 2 at around 450.5s
then finished generating at 765.  

So this is an improvement. 
I pulled these test prompt form some forum someplace, but maybe there are other better ones.

**Comment by nathanlesage** [comment]
Regarding the slow speed: I haven't used an i5 in quite some time, so I don't know if these numbers are odd or to be expected. But it's definitely not decent. I haven't gotten a great idea of how to implement configuration, but once I do I'll enable the possibility to switch to the CUDA version of llama.cpp, which should increase inference speed drastically.

**Comment by Willxiam** [comment]
This is the reason I put older machine in the thread.  I do not know either.  But I will keep leaving feedback because I suspect many others will be trying to figure out the same.  Thanks for all of your help.  I will try to test out the models on the newer machine I have tomorrow.

**Comment by nathanlesage** [comment]
I'm gonna simply leave the same notice I have just added to the README here for your information:

> I noticed that LocalChat is gaining in popularity. That's great! However, this year I will have to defend my PhD thesis, so I will have limited time to code, which I will mainly direct towards my main app, [Zettlr](https://github.com/Zettlr/Zettlr). This means that LocalChat won't receive many commits from my side. That being said, **it is much easier to review & merge pull requests** and in case there are PRs that improve the app, I'll happily review and merge, and continue to update the app. However, this will only work if people open PRs. So, if you enjoy the app, please contribute. Regular development on my own side can proceed in early 2026 again.
